### DISCUSSION TITLE
Websocket client fails to connect.

### DISCUSSION THREAD

**Discussion by parasharsangeet** [cesanta/mongoose#1625] [Ask questions, get answers] [answered]
I was trying to connect to the discord gateway using the websocket url which is provided by the documentation. However, for some reason i get a DNS error when i try to run the code. I have attached the code along with the output for convenience.

 ![image](https://user-images.githubusercontent.com/91406232/177979241-8b9dde36-85b2-47cd-a2eb-be11e09c3f48.png)
![image]( https://user-images.githubusercontent.com/91406232/177978896-edcb2c24-bbad-4b84-9f17-48764bf0bfc5.png)

**Comment by scaprile** [comment, marked as answer]
Did you read the [tutorial](https://mongoose.ws/tutorials/websocket-client) ? Did you try the [example code](https://github.com/cesanta/mongoose/examples/websocket-client) ? 
You don't seem to be calling the event manager, is that your actual code ? Please read and do the tutorials, and work your way up from there.
Further on up your road:
- you are providing a TLS URL and you are not initializing TLS, please read the [TLS tutorial](https://mongoose.ws/tutorials/tls/) and have a non-TLS client working before you dive into TLS.
- you are not upgrading your connection to WS
